services/server/src/server/session.ts

```diff
@@ -0,0 +1,115 @@
+import expressSession from "express-session";
+import createMemoryStore from "memorystore";
+import { Pool } from "pg";
+import genFunc from "connect-pg-simple";
+import config from "config";
+import session from "express-session";
+
+import logger from "../common/logger";
+import { NextFunction, Request, Response } from "express";
+
+function initMemoryStore() {
+  const MemoryStore = createMemoryStore(expressSession);
+
+  logger.info("Using memory based session");
+  return new MemoryStore({
+    checkPeriod: config.get("session.maxAge"),
+  });
+}
+
+function initDatabaseStore() {
+  const pool = new Pool({
+    host: process.env.SOURCIFY_POSTGRES_HOST,
+    database: process.env.SOURCIFY_POSTGRES_DB,
+    user: process.env.SOURCIFY_POSTGRES_USER,
+    password: process.env.SOURCIFY_POSTGRES_PASSWORD,
+    port: parseInt(process.env.SOURCIFY_POSTGRES_PORT || "5432"),
+  });
+
+  // This listener is necessary otherwise the sourcify process crashes if the database is closed
```

Comment:
Isn't this something expected? If the database is closed, that means we can't write anything, and it should really crash?

Reply:
Crashes with a bad, not understandable error, we can throw inside this listener a better error if that's what we want

Reply:
It's ok to add a human readable error message, alongside what we receive from Postgres but 

1) The postgres message should be available. Isn't completely useless?
2) It should indeed crash. My understanding from the error listener below is that Sourcify will not crash. Is it going to crash? (It should crash)